sql/core/src/main/scala/org/apache/spark/sql/execution/joins/CartesianProduct.scala

```diff
@@ -28,9 +28,17 @@ import org.apache.spark.sql.execution.metric.SQLMetrics
  * :: DeveloperApi ::
  */
 @DeveloperApi
-case class CartesianProduct(left: SparkPlan, right: SparkPlan) extends BinaryNode {
+case class CartesianProduct(
+    left: SparkPlan,
+    right: SparkPlan,
+    buildSide: BuildSide) extends BinaryNode {
```

Comment:
I am not so sure if the change necessary, as if either side of the table is small enough, we will resort to the `BroadcastNestedLoopJoin`; since both side tables are bigger enough, the change may not improve the performance as we expected.
BTW, we didn't build hash map in CaresianProduct, using `buildSide` seems a little confusing.